torch/distributed/_tensor/_dispatch.py

```diff
@@ -307,15 +308,22 @@ def unwrap_to_op_info(
         local_kwargs: Dict[str, object] = {}
         mesh: Optional[DeviceMesh] = None
 
+        # record the largest mesh only if implicit replication is turned on
+        if self._allow_implicit_replication:
+            for arg in chain(iter(args_list), iter(kwargs.values())):
+                if isinstance(arg, dtensor.DTensor):
+                    if mesh is None or arg.device_mesh.ndim > mesh.ndim:
+                        mesh = arg.device_mesh
+                    # Currently, we only support implicit replication from 1D to 2D for partial TP
+                    # use cases when using 2D. Therefore, we break out the loop when we find a 2D mesh.
+                    # If we see more generic nD use case, we should remove the constraint.
+                    elif mesh.ndim == 2:
```

Comment:
hmmm I feel this is too restrictive and should not be put into the general dispatching logic, it should be put into the implicit replicate spec function at least

Reply:
Moved this to a separate util and replaced the early break constraint to be if the mesh we found is already a root mesh. This would be more generic. The only caveat is that for 3D, if using partial TP, it would loop through the entire list for foreach.